File: aexample/src/main/java/org/judal/examples/java/jdbc/E03_ExecuteDDLOnRelationalDataSource.java
package org.judal.examples.java.jdbc;

import org.junit.Test;
import static org.junit.Assert.assertTrue;


import org.judal.jdbc.JDBCRelationalDataSource;

/**
 * Example of how to execute any SQL statement over a JDBC data source
 */
public class E03_ExecuteDDLOnRelationalDataSource {

	@Test
	public void demo() throws Exception {
		JDBCRelationalDataSource dataSource = E02_CreateAdditionalDataSource.create();
		
		dataSource.execute("CREATE TABLE test_table (column1 INTEGER, column2 VARCHAR(100))");
		
		// Model changes won't be visible until the data sources is restarted
		dataSource.restart();
		
		assertTrue(dataSource.exists("test_table", "U"));
		
		dataSource.execute("DROP TABLE test_table");
		
		E02_CreateAdditionalDataSource.close(dataSource);
	}
	
}

File: aexample/src/main/scala/org/judal/examples/scala/jdbc/E03_ExecuteDDLOnRelationalDataSource.scala
package org.judal.examples.scala.jdbc

import org.junit.Test
import org.junit.Assert.assertTrue

import org.scalatest.Suite

import org.judal.jdbc.JDBCRelationalDataSource

/**
 * Example of how to execute any SQL statement over a JDBC data source
 */
class E03_ExecuteDDLOnRelationalDataSource extends Suite {

	@Test def demo() = {
		val dataSource = E02_CreateAdditionalDataSource.create
		
		dataSource.execute ("CREATE TABLE test_table (column1 INTEGER, column2 VARCHAR(100))")
		
		// Model changes won't be visible until the data sources is restarted
		dataSource.restart
		
		assertTrue (dataSource.exists("test_table", "U"))
		
		dataSource.execute ("DROP TABLE test_table")
		
		E02_CreateAdditionalDataSource.close (dataSource)
	}
	
}
